### PR TITLE
Refactor Section Outbox Work Item Partition Usage

### PR DESCRIPTION
**Pull request by mmarcotte**
Solves: https://github.com/ustaxcourt/ef-cms/issues/1739

Need to run the script in order to move all of the records to the correct partition in order to clean up the existing partitions. And then future records should all expire properly. We can make use of the archive data with future stories.

This refactor continues to store records on the original primary partition of `section-outbox|${section}`, but it adds a `ttl` so these partitions don't collect data forever and become overloaded.

And it also stores a copy of the record in a partition based on the month `section-outbox|${section}|{YYYY-MM}` of the `sk`. This way we can query for records if we are attempting to go back in time further than 7 days. This functionality isn't written yet, but we didn't want to lose any data in case we wanted to do this at a later date.

Additionally, it makes use of the handy `deleteByGsi(...)` persistence helper function so that we can easily remove all of the records associated with a Work Item when we need to delete it.

### CODE REVIEW

File: shared/src/persistence/dynamo/workitems/createSectionOutboxRecords.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      Item,
+      applicationContext,
+      section,
+    }),
+    await createSectionOutboxArchiveRecord({
+      Item,
+      applicationContext,
+      section,
+    }),
+  ]);
```
Comment (kkoskelin) [inline shared/src/persistence/dynamo/workitems/createSectionOutboxRecords.js]: you can actually
1. remove the `async` decorator from the function definition
2. remove all `await` decorators
3. directly return the value from `Promise.all(...)`

File: shared/src/persistence/dynamo/workitems/deleteWorkItem.js [hunk 2]
```diff
@@ -1,10 +1,4 @@
...
 
-exports.deleteWorkItem = ({ applicationContext, workItem }) =>
-  client.delete({
-    applicationContext,
-    key: {
-      pk: `case|${workItem.docketNumber}`,
-      sk: `work-item|${workItem.workItemId}`,
-    },
-  });
+exports.deleteWorkItem = async ({ applicationContext, workItem }) =>
```
Comment (kkoskelin) [inline shared/src/persistence/dynamo/workitems/deleteWorkItem.js]: `async` decorator can be removed

File: shared/src/persistence/dynamo/workitems/createSectionOutboxRecords.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  workItem,
+}) => {
+  const Item = {
+    ...workItem,
+    gsi1pk: `work-item|${workItem.workItemId}`,
+    sk: workItem.completedAt ? workItem.completedAt : workItem.updatedAt,
+  };
+
+  await Promise.all([
+    await createSectionOutboxRecentRecord({
```
Comment (codyseibert) [inline shared/src/persistence/dynamo/workitems/createSectionOutboxRecords.js]: technically you don't need these inner await calls.

File: shared/src/persistence/dynamo/workitems/createSectionOutboxRecords.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * @param {Object} providers The providers object
+ * @param {Object} providers.applicationContext The application context
+ * @param {String} section docket or petitions
+ * @param {WorkItem} workItem The instantiated Work Item to create a record
+ * @returns {Promise} resolves upon completion of persistence request
+ */
+const createSectionOutboxArchiveRecord = async ({
+  applicationContext,
+  Item,
```
Comment (codyseibert) [inline shared/src/persistence/dynamo/workitems/createSectionOutboxRecords.js:30]: I find it strange that this is `Item` and not lower case i.  Most properties on our JS objects are all camalCase, but maybe there are a few places we just need to go back and refactor.

Comment (mmarcotte) [reply shared/src/persistence/dynamo/workitems/createSectionOutboxRecords.js:30]: Yeah, put takes `Item` for the dynamoDB call.

**Review by kkoskelin — APPROVED** [review]

**Review by mark-meyer — APPROVED** [review]

**Pull request merged**